### DISCUSSION TITLE
Is there a way to retreive `password complexity settings` without `ORG_SETTINGS_MANAGER` ?

### DISCUSSION THREAD

**Discussion by Yberion** [zitadel/zitadel#7917] [Q&A] [answered]
Hello,

I'm implementing myself a password change on my Front-End and currently, only users with the role `ORG_SETTINGS_MANAGER` seems to be allowed to query the password complexity settings.

Also, currently ORG roles can't be set automatically and anyway I don't want to give `ORG_SETTINGS_MANAGER` as it will give way more permissions than necessary https://github.com/zitadel/zitadel/blob/main/cmd/defaults.yaml#L1252.

So, is there a way to get the password complexity settings with the user's token but without `ORG_SETTINGS_MANAGER` ?

This will allow me to make a request to Zitadel, get the settings and create Angular Validators depending on the response of Zitadel without hardcoding anything and risking to be desynchronized.

This is why I asked this question https://github.com/zitadel/zitadel/discussions/7916 as for now I'm forced to hardcode the regex on my Front-End.

When registering a new user, as this is a Zitadel page, I think you retreive infos from somewhere, but we can't if we make our own UI in our Front-End (unless I don't know how).

![image](https://github.com/zitadel/zitadel/assets/4186385/17bc068c-1b7b-472c-a73c-ba36f6c8f10a)

**Comment by hifabienne** [comment, marked as answer]
Hei @Yberion 
With the auth api, you are able to make requests with the currently authenticated user.
We have implemented the get Requests for some settings, which should allow you to do exactly what you like to:
https://zitadel.com/docs/category/apis/resources/auth/policies

You can find the get passwordcomplexity policy in there

**Reply by Yberion** [reply]
Hello @hifabienne,

Damn sorry, I did only check the v2 API, thanks for the answer!

Will this passwordcomplexity policy be available on v2beta in the future? (In case v2 is stable and v1 is deprecated)

**Reply by hifabienne** [reply]
Yes the goal for the future is to have all the request we currently have also in the new apis. But it will take some time to get there
